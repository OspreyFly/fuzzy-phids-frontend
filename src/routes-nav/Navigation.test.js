import React from 'react';
import { render } from '@testing-library/react';
import Navigation from './Navbar';


const currentUser = {id: 1, username: "testuser"};

jest.mock('react-router-dom', () => ({
  NavLink: ({ children }) => <div>{children}</div>,
  Link: ({ onClick }) => {
    const handleClick = jest.fn(onClick);
    return <div onClick={handleClick}>Link</div>;
  },
  useNavigate: jest.fn(),
}));

test('matches snapshot', () => {
  const { asFragment } = render(
    <Navigation currentUser={currentUser} />
  );
  expect(asFragment()).toMatchSnapshot();
});