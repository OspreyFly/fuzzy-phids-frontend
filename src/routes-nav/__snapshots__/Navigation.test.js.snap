// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      id="title"
    >
      Sid's Fuzzy Phids
    </h1>
    <nav
      id="nav-container"
    >
      <ul
        id="nav-list"
      >
        <li>
          Home
        </li>
        <li>
          <div>
            Browse
          </div>
        </li>
        <li>
          <div>
            Check-Out
          </div>
        </li>
        <li>
          <div>
            Link
          </div>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
